src/main/java/org/tikv/common/PDClient.java

```diff
@@ -341,6 +341,27 @@ private boolean isScatterRegionFinish(GetOperatorResponse resp) {
     return new Pair<Metapb.Region, Metapb.Peer>(decodeRegion(resp.getRegion()), resp.getLeader());
   }
 
+  @Override
+  public List<Pdpb.Region> scanRegions(
+      BackOffer backOffer, ByteString startKey, ByteString endKey, int limit) {
+    // no need to backoff because ScanRegions is just for optimization
+    // introduce a warm-up timeout for ScanRegions requests
+    PDGrpc.PDBlockingStub stub =
+        getBlockingStub().withDeadlineAfter(conf.getWarmUpTimeout(), TimeUnit.MILLISECONDS);
+    Pdpb.ScanRegionsRequest request =
```

Comment:
should we use `callWithRetry` here?

Reply:
I don't think we should use callWithRetry here because we don't need to retry when the request fails.

Reply:
then i suggest to rename `scanRegions` to `scanRegionsWithoutRetry`.
The gRPC methods do retry as default.